File: code-examples/scala/authentication/default.scala
import org.mongodb.scala._

object MongoDBConnection {
  // Replace with values for your environment.
  val username: String = "<username>"
  val password: String = "<password>"
  val clusterUrl: String = "<MongoDB cluster url>"

  // Connection related configurations
  val settings: MongoClientSettings = MongoClientSettings.builder()
    .applyConnectionString(
      ConnectionString(
        s"mongodb+srv://${java.net.URLEncoder.encode(username, "UTF-8")}:${java.net.URLEncoder.encode(password, "UTF-8")}@$clusterUrl/?authMechanism=DEFAULT"
      )
    )
    .build()

  // Create a new MongoClient
  val mongoClient: MongoClient = MongoClient(settings)

  // Function to connect to the server
  def connectToDB(): Unit = {
    try {
      // Establish and verify the connection
      mongoClient
        .getDatabase("admin")
        .runCommand(Document("ping" -> 1))
        .printHeadResult("Connected successfully to server")
    } finally {
      // Ensures that the client will close when you finish/error
      mongoClient.close()
    }
  }
  def main(args: Array[String]): Unit = {
    try {
      connectToDB()
    } catch {
      case e: Exception =>
        println(e.printStackTrace())
    }
  }
}

File: code-examples/php/authentication/default.php
<?php

require 'vendor/autoload.php';

$username = urlencode("<username>");
$password = urlencode("<password>");
$clusterUrl = "<MongoDB cluster url>";

$uri = "mongodb+srv://{$username}:{$password}@{$clusterUrl}?authSource=admin"; 

// Create a new MongoClient
$client = new MongoDB\Client($uri);

// Function to connect to the server
function run() 
{
    global $client;

    try {
        // Establish and verify connection
        $command = new MongoDB\Driver\Command(['ping' => 1]);
        $client->getManager()->executeCommand('admin', $command);

        echo "Connected successfully to server.\n";
    } catch (Exception $e) {
        echo $e->getMessage(), "\n";
        exit;
    } finally {
        // Ensures that the client will close when you finish/error
        $client = null;
    }
}

run();
